src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -138,30 +138,43 @@
             ArgumentTypeInfos.registerByClass(ModIdArgument.class,
                     SingletonArgumentInfo.contextFree(ModIdArgument::modIdArgument)));
 
-    public static final RegistryObject<Attribute> SWIM_SPEED = ATTRIBUTES.register("swim_speed", () -> new RangedAttribute("forge.swimSpeed", 1.0D, 0.0D, 1024.0D).setSyncable(true));
-    public static final RegistryObject<Attribute> NAMETAG_DISTANCE = ATTRIBUTES.register("nametag_distance", () -> new RangedAttribute("forge.nameTagDistance", 64.0D, 0.0D, 64.0).setSyncable(true));
+    public static final RegistryObject<Attribute> SWIM_SPEED = ATTRIBUTES.register("swim_speed", () -> new RangedAttribute("forge.swim_speed", 1.0D, 0.0D, 1024.0D).setSyncable(true));
+    public static final RegistryObject<Attribute> NAMETAG_DISTANCE = ATTRIBUTES.register("nametag_distance", () -> new RangedAttribute("forge.name_tag_distance", 64.0D, 0.0D, 64.0).setSyncable(true));
     public static final RegistryObject<Attribute> ENTITY_GRAVITY = ATTRIBUTES.register("entity_gravity", () -> new RangedAttribute("forge.entity_gravity", 0.08D, -8.0D, 8.0D).setSyncable(true));
 
     /**
-     * Reach Distance represents the distance at which a player may interact with the world.  The default is 4.5 blocks.  Players in creative mode have an additional 0.5 blocks of reach distance.
-     * @see IForgePlayer#getReachDistance()
-     * @see IForgePlayer#canInteractWith(BlockPos, double)
-     * @see IForgePlayer#canInteractWith(Entity, double)
+     * Reach Distance represents the distance at which a player may interact with the world.  The default is 4.5 blocks.  Players in creative mode have an additional 0.5 blocks of block reach.
+     * @see IForgePlayer#getBlockReach()
+     * @see IForgePlayer#canReach(BlockPos, double)
      */
-    public static final RegistryObject<Attribute> REACH_DISTANCE = ATTRIBUTES.register("reach_distance", () -> new RangedAttribute("generic.reachDistance", 4.5D, 0.0D, 1024.0D).setSyncable(true));
+    public static final RegistryObject<Attribute> BLOCK_REACH = ATTRIBUTES.register("block_reach", () -> new RangedAttribute("forge.block_reach", 4.5D, 0.0D, 1024.0D).setSyncable(true));
 
     /**
-     * Attack Range represents the distance at which a player may attack an entity.  The default is 3 blocks.  Players in creative mode have an additional 3 blocks of attack reach.
-     * @see IForgePlayer#getAttackRange()
-     * @see IForgePlayer#canHit(Entity, double)
+     * TODO: Remove
+     * @deprecated - use {@link #BLOCK_REACH}
      */
-    public static final RegistryObject<Attribute> ATTACK_RANGE = ATTRIBUTES.register("attack_range", () -> new RangedAttribute("generic.attack_range", 3.0D, 0.0D, 1024.0D).setSyncable(true));
+    public static final RegistryObject<Attribute> REACH_DISTANCE = BLOCK_REACH;
```

Comment:
Shouldn't this be annotated `@Deprecated` instead of just Javadoc deprecated? My compiler always yells at me if I annotate something only through the javadoc.

Reply:
Maybe, eclipse was smart enough to detect it through the jdoc, and IDEA tends to be "smarter" than eclipse, so I figured it was sufficient.

Reply:
We use the deprecated annotation, and the since value to tell us when to delete.